datadog_checks_base/tests/base/utils/db/test_query_executor.py

```diff
@@ -149,3 +153,53 @@ def test_many_queries(self, aggregator):
 
         for i in range(num_queries):
             aggregator.assert_metric('test.metric.{}'.format(i), i, metric_type=aggregator.GAUGE, tags=tags)
+
+    @pytest.mark.parametrize(
+        'collection_interval, expected_exception',
+        [
+            pytest.param(1, None, id='valid interval 1s'),
+            pytest.param(0.5, ValueError, id='invalid interval 0.5s'),  # 0.5s is invalid because it rounds to 0
+            pytest.param(0, ValueError, id='invalid interval 0s'),
+            pytest.param(-1, ValueError, id='invalid interval -1s'),
+            pytest.param('test', ValueError, id='invalid interval not a number'),
+        ],
+    )
+    def test_query_with_collection_interval(self, aggregator, collection_interval, expected_exception):
+        """Test running a query with a custom collection interval"""
+        queries = [
+            {
+                'name': 'query1',
+                'query': 'select 1',
+                'columns': [{'name': 'test.metric_with_interval', 'type': 'gauge'}],
+                'collection_interval': collection_interval,
+            }
+        ]
+
+        check = AgentCheck('test', {}, [{}])
+        qe = QueryExecutor(mock_executor([[1]]), check, queries)
+
+        if expected_exception:
+            with pytest.raises(expected_exception):
+                qe.compile_queries()
```

Comment:
```suggestion
          with nullcontext() if expected_exception is None else pytest.raises(expected_exception):
              qe.compile_queries()
```

Based on [these docs](https://docs.pytest.org/en/8.0.x/example/parametrize.html#parametrizing-conditional-raising), expects you import `nullcontext`.

Reply:
Oh true! 

tbh, I'd probably split this up into 2 tests: one with the happy path and one with all the different errors. Whatever shared code remains you can put into helper functions.